### PR TITLE
chore(format): add prettier to project 

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue URL: #285 

## What is the new behavior?

this commit reworks the pre-existing prettier infrastructure in the
project in an attempt to add continuous integration (ci) checks for
formatting to the project.

prettier and ionic's prettier configuration are added as a top level
dependencies to the project, such that lerna can format all projects
with a 'prettier' script in the future. by using ionic's prettier
configuration, we can converge upon previously agreed upon formatting
config.

ci is updated to run prettier in parallel with other jobs, allowing us
to 'fail-fast' in the event of a formatting error. there is potential
room in the future to refactor ci jobs/steps by adding this parallelism,
but that was not done here in order to limit the scope of this commit.

this commit only handles the angular output target. this allows for
incremental improvement/proof-of-concept to get the idea out to the
teams managing this project for feedback sooner than later

format the contents of the angular output target using the configuration
added in the previous commit.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Looking for feedback/thoughts here from folks more familiar with Lerna setups and whatnot.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,21 +8,25 @@
...
   "devDependencies": {
+    "@ionic/prettier-config": "^2.0.0",
     "@stencil/core": "^2.9.0",
     "@types/jest": "^26.0.9",
     "jest": "^26.3.0",
     "lerna": "^4.0.0",
     "lerna-changelog": "^1.0.1",
     "np": "^6.0.2",
-    "prettier": "^2.0.5",
+    "prettier": "2.7.1",
```
Comment (rwaskiewicz) [inline package.json:24]: Locking this as Prettier doesn't necessarily follow semver

File: package.json [hunk 2]
```diff
@@ -8,21 +8,25 @@
...
     "rimraf": "^2.6.3",
     "rollup": "^2.23.1",
+    "semver": "^5.5.0",
```
Comment (rwaskiewicz) [inline package.json:27]: This moved as a result of adding new dependencies via `npm`'s CLI - might as well keep it 🤷

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: 16
+          cache: 'npm'
+
+      - name: Install Dependencies
+        run: npm ci
+        shell: bash
+
+      - name: Prettier Check
+        run: npm run prettier.dry-run
```
Comment (liamdebeasi) [inline .github/workflows/format.yml:27]: Is this supposed to fail if prettier was not run?

Comment (rwaskiewicz) [reply .github/workflows/format.yml:27]: Yep! The `dry-run` command runs `prettier --list-different`, which gives a non-zero exit code if any non-formatted files are detected

File: packages/angular-output-target/src/generate-angular-component.ts [hunk 4]
```diff
@@ -1,130 +1,128 @@
...
-      // `outputReferenceRemap` should be updated only if the import interface is set in outputsInterface,
-      // this will prevent global types to be remapped.
-      const remappedReference = `I${cmpMeta.componentClassName}${reference}`;
-      if (refObject.location === 'local' || refObject.location === 'import') {
-        outputReferenceRemap[reference] = remappedReference;
-        let importLocation: string = componentCorePackage;
-        if (componentCorePackage !== undefined) {
-          const dirPath = includeImportCustomElements ? `/${customElementsDir || 'components'}` : '';
-          importLocation = `${normalizePath(componentCorePackage)}${dirPath}`;
+export const createComponentDefinition =
```
Comment (rwaskiewicz) [inline packages/angular-output-target/src/generate-angular-component.ts:4]: These changes are a little more palatable if you "ignore whitespace" in GH's diff view

**Review by liamdebeasi — APPROVED** [review]

**Review by tanner-reits — APPROVED** [review]

File: packages/angular-output-target/package.json [hunk 5]
```diff
@@ -20,6 +20,9 @@
     "watch": "tsc --watch",
     "rollup": "rollup -c",
     "version": "npm run build",
+    "prettier": "npm run prettier.base -- --write",
+    "prettier.base": "prettier \"./({anguler-output-target,src,test}/**/*.{ts,tsx,js,jsx})|*.{ts,tsx,js,jsx}\"",
```
Comment (rwaskiewicz) [inline packages/angular-output-target/package.json:24]: Oops - I need to change this - should be 'angular-component-lib'

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

**Pull request merged**